.github/workflows/tests.yml

```diff
@@ -17,23 +17,15 @@ jobs:
   tap_syntax:
     if: github.repository == 'Homebrew/homebrew-core'
     runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/homebrew/ubuntu16.04:master
```

Comment:
Can we do the same for the `tests` step? Though that might now put us back into duplicating all the tests steps again...

Reply:
Nooo, we simply cannot because of Actions syntax limitation. `container` key can't be specified when runner OS is macOS. It destroys the whole workflow run.

Reply:
Mainly the error will be:
```
Error: File not found: 'docker'
```

Reply:
A draft PR to see what it'd look like if we split the steps entirely for macOS and Linux would be interesting?

Reply:
What do you mean exactly? Split macOS and Linux into separate jobs?

Reply:
Yeh, different jobs but in the same file. I'd be interested what that looks like.

Reply:
Might do in my spare time as it interests me as well.